lib/rules/string-no-newline/__tests__/index.mjs

```diff
@@ -145,3 +145,93 @@ testRule({
 		},
 	],
 });
+
+testRule({
+	ruleName,
+	config: [true, { ignore: ['at-rules'] }],
+
+	accept: [
+		{
+			code: "@import 'foo\n.css';",
```

Comment:
note
nit

I suppose that `string` in the rule name renders us unable to reject `@import url(foo\n.css);`—without `ignore: ['at-rules']`—due to the fact that the url is not surrounded by quotes; even if it would functionally be the same.
ref https://drafts.csswg.org/css-syntax-3/#consume-url-token

Reply:
Yes, it's out of the scope of this rule.

Two other rule will be or are active in this area, though:

- `function-url-quotes` can be used to enforce quotes in `url()`
- `at-rule-prelude-no-invalid` will catch `@import url(foo\n.css);`

As this rule doesn't target unquoted URLs, there's no overlap with the upcoming `at-rule-prelude-no-invalid` rule.